### PR TITLE
hide jwt and saml when disabled

### PR DESCRIPTION
**Pull request by alxnddr**
Related to https://github.com/metabase/metabase-private/issues/82 and https://github.com/metabase/metabase-private/issues/83

### Description

Hides JWT and SAML settings when `sso` token feature disabled.

It is more convenient to review this PR with the "Hide whitespace" setting enabled:
Screen Shot 2023-07-05 at 9 19 50 PM

#### OSS build
- Login as an admin
- Admin -> Settings -> Authentication
- Ensure there is no "SAML" and "JWT" options
- Ensure http://localhost:3000/admin/settings/authentication/saml direct link does not show SAML settings
- Ensure http://localhost:3000/admin/settings/authentication/jwt direct link does not show JWT settings

#### EE build without token
- Login as an admin
- Admin -> Settings -> Authentication
- Ensure there is no "SAML" and "JWT" options
- Ensure http://localhost:3000/admin/settings/authentication/saml direct link does not show SAML settings
- Ensure http://localhost:3000/admin/settings/authentication/jwt direct link does not show JWT settings

#### EE build with token
- Login as an admin
- Admin -> Settings -> Authentication
- Ensure there are "SAML" and "JWT" options
- Ensure the settings pages for both work as expected

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/auth/index.js [hunk 1]
```diff
@@ -21,246 +21,246 @@ import { SsoButton } from "./components/SsoButton";
...
-    },
-  ]),
-);
-
-PLUGIN_ADMIN_SETTINGS_UPDATES.push(sections => ({
-  ...sections,
-  "authentication/saml": {
-    component: SettingsSAMLForm,
-    settings: [
+if (hasPremiumFeature("sso")) {
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/auth/index.js:24]: will there be a `sso_legacy` flag?

File: frontend/src/metabase/admin/settings/containers/SettingsEditorApp/index.ts
Comment (iethree) [inline frontend/src/metabase/admin/settings/containers/SettingsEditorApp/index.ts:1]: I think you need 
`// eslint-disable-next-line import/no-default-export -- deprecated usage`

File: frontend/src/metabase/admin/settings/selectors.js [hunk 2]
```diff
@@ -67,642 +67,661 @@ function updateSectionsWithPlugins(sections) {
...
-      },
-      {
-        key: "humanization-strategy",
-        display_name: t`Friendly Table and Field Names`,
-        type: "select",
-        options: [
-          {
-            value: "simple",
-            name: t`Replace underscores and dashes with spaces`,
+const buildSections = _.memoize(() =>
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/selectors.js:70]: ❤️

**Review by iethree — APPROVED** [review]
LGTM

File: enterprise/frontend/src/metabase-enterprise/auth/index.js (hunk 1 above)
Comment (alxnddr) [reply enterprise/frontend/src/metabase-enterprise/auth/index.js:24]: It is not clear at the moment, I'll update after the decision has been made. Probably, only a part of this huge `if` will be available

**Comment by alxnddr** [comment]
Too many conflicts & requirements has changed. Reworking

**Pull request closed without merging**